system/CodeIgniter.php

```diff
@@ -290,6 +300,8 @@ protected function initializeKint()
      */
     public function run(?RouteCollectionInterface $routes = null, bool $returnResponse = false)
     {
+        assert($this->context !== null, 'Context must be set before run() is called');
```

Comment:
is this will cause warning when user run `composer update` without update public/index.php?

Reply:
Yes, I added upgrade_420.rst.

Reply:
Probably update the message? like:

```suggestion
assert(
    $this->context !== null, 
    'Context must be set before run() is called, if you are upgrading from 4.1.x, you can merge public/index.php and spark file from vendor/codeigniter4/framework'
);
```

Reply:
Updated.